File: tests/new_tests/sealed_copy.jl
struct Packet
    val::Float64
end

struct Register
    PACKET::Packet
    VALUE::Int64
end

PACKET::Packet = None
VALUE::Int64 = None

function main()
    a = VALUE(Register, 10)
    @assert(is_value(a))
    value(a);
    b = PACKET(Register, Packet(1.3))
    @assert(is_packet(b))
    packet(b);
    println(join(["OK"], " "));
end

main()

File: pyjl/tests/sealed_copy.py
# On hold until @sealed is solved
from adt import adt as sealed
from dataclasses import dataclass


@dataclass
class Packet:
    val: float

@sealed
class Register:
    PACKET: Packet
    VALUE: int


if __name__ == "__main__":
    a = Register.VALUE(10)
    assert a.is_value()
    a.value()
    b = Register.PACKET(Packet(1.3))
    assert b.is_packet()
    b.packet()
    print("OK")

# Copied from 'cases' folder